### ISSUE TITLE
Assertion failed: sender == m_clients.find(clipboard.m_clipboardOwner)->second

### ISSUE THREAD

**Issue by rubycoder** [input-leap/input-leap#1593]
### What happened?

About once a day, on Windows 7 SP1, I see:

Assertion failed: sender == m_clients.find(clipboard.m_clipboardOwner)->second, file D:\a\input-leap\input-leap\src\lib\server\Server.cpp, line 1457

### Version

v2.4.0

### If applicable, where did you install Input Leap from?

https://github.com/input-leap/input-leap/actions/runs/3896229827

### What OSes are you seeing the problem on? (Check all that apply)

Windows

### What OS versions are you using?

Windows 7 SP 1 - server
macOS 12.6.2 - client

### Relevant log output

```shell
[2023-01-20T15:37:03] INFO: switch from "P50" to "M" at 0,154
	D:\a\input-leap\input-leap\src\lib\server\Server.cpp,409
[2023-01-20T15:37:03] INFO: leaving screen
	D:\a\input-leap\input-leap\src\lib\inputleap\Screen.cpp,130
[2023-01-20T15:38:15] INFO: switch from "M" to "P50" at 1845,296
	D:\a\input-leap\input-leap\src\lib\server\Server.cpp,409
[2023-01-20T15:38:15] INFO: entering screen
	D:\a\input-leap\input-leap\src\lib\inputleap\Screen.cpp,112
A
ssertion failed: sender == m_clients.find(clipboard.m_clipboardOwner)->secon
d, file D:\a\input-leap\input-leap\src\lib\server\Server.cpp, line 1457
[2023-01-20T15:52:35] WARNING: detected application not running, pid=45124
[2023-01-20T15:52:36] INFO: backing off, wait=2s, failures=1
[2023-01-20T15:52:38] INFO: starting new process
[2023-01-20T15:52:39] INFO: activeDesktop:Default
[2023-01-20T15:52:39] INFO: starting new process
[2023-01-20T15:52:39] INFO: drag and drop enabled
	D:\a\input-leap\input-leap\src\lib\inputleap\App.cpp,174
started server (IPv4/IPv6), waiting for clients
[2023-01-20T15:52:39] INFO: service status: active
[2023-01-20T15:52:40] INFO: OpenSSL 1.0.2l  25 May 2017
	D:\a\input-leap\input-leap\src\lib\net\SecureSocket.cpp,849
[2023-01-20T15:52:41] INFO: accepted secure socket
	D:\a\input-leap\input-leap\src\lib\net\SecureSocket.cpp,472
[2023-01-20T15:52:41] INFO: AES256-GCM-SHA384       TLSv1.2 Kx=RSA      Au=RSA  Enc=AESGCM(256) Mac=AEAD
	D:\a\input-leap\input-leap\src\lib\net\SecureSocket.cpp,867
[2023-01-20T15:52:41] NOTE: accepted client connection
	D:\a\input-leap\input-leap\src\lib\server\ClientListener.cpp,143
[2023-01-20T15:52:41] NOTE: client "M" has connected
	D:\a\input-leap\input-leap\src\lib\server\Server.cpp,289
```

**Comment by shymega** [comment]
I don't think we support windows 7 officially, but there's a lot of changes happening on master right now, so you probably shouldn't use the builds at the moment. Not until we release version v3.0.0. 

I will copy @p12tic into this as it's related to the clipboard change made recently.

**Comment by rubycoder** [comment]
Thanks for your quick response!

>> I don't think we support windows 7 officially

Perhaps I misinterpreted the FAQ:

Q: What OSes are supported?

A: The most recent release of Input Leap is known to work on:

Windows 7, 8, 8.1, 10, and 11
macOS ...

>> Not until we release version v3.0.0.

Any rough estimate of when v3.0.0 might be?

**Comment by shymega** [comment]
Oh right, I'm thinking of XP. Sorry, I replied late at night (and on GitHub mobile!), but yes, 7 is fine. I believe, @p12tic is heavily refactoring Input Leap and modernising it, so older OS versions would be best to use Synergy 1.x, but it [using Input Leap] comes with the benefit of better memory management, event handling improvements, easier to-work with the codebase.

v3.0.0 is still a WIP. Not sure when yet. Have you tried the latest commit from Actions/Azure?

**Comment by vancluever** [comment]
@shymega I think this possibly related to #1248 and could do with some more investigation. As mentioned there if you just turn the assertion into a guard (basically just flip the condition, log, and reject the change when it triggers, it seems to correct this issue outright, at the tradeoff of possibly missing slower clipboard updates when focus is switched back quickly. I can put in a PR, but I'm not too sure if ya'll have possibly a better idea of how to solve it given the possibility of missing some updates as mentioned in my [other comment](https://github.com/input-leap/input-leap/issues/1248#issuecomment-1475477521).

**Comment by vancluever** [comment]
I've put in the PR for this as from what I've looked at there honestly does not seem to be much of a better way to go about fixing this, for now. I feel like ultimately there would need to be some re-working of clipboard sharing to allow for off-screen clipboard updates, but again, I may have missed something.
